modules/gapi/test/rmat/rmat_view_tests.cpp

```diff
@@ -0,0 +1,147 @@
+    EXPECT_EQ(view.ptr(), mat.ptr());
+}
+
+TEST_P(RMatViewTest, DefaultStep) {
+    auto type = GetParam();
+    GMatDesc desc;
+    desc.chan = CV_MAT_CN(type);
+    desc.depth = CV_MAT_DEPTH(type);
+    desc.size = {8,8};
+    std::vector<unsigned char> data(desc.size.width*desc.size.height*CV_ELEM_SIZE(type));
+    View view(desc, data.data());
+    EXPECT_EQ(view.step(), static_cast<size_t>(desc.size.width)*CV_ELEM_SIZE(type));
+}
+
+static Mat asMat(View& view) {
+    return Mat(view.size(), view.type(), view.ptr(), view.step());
+}
+
+TEST_P(RMatViewTest, NonDefaultStepInput) {
+    auto type = GetParam();
+    Mat bigMat(16,16,type);
+    cv::randn(bigMat, cv::Scalar::all(127), cv::Scalar::all(40));
+    Mat mat = bigMat(cv::Rect{4,4,8,8});
+    View view(cv::descr_of(mat), mat.data, mat.step);
+    const auto viewMat = asMat(view);
+    Mat ref, out;
+    cv::Size ksize{1,1};
+    cv::blur(viewMat, out, ksize);
+    cv::blur(    mat, ref, ksize);
+    EXPECT_EQ(0, cvtest::norm(ref, out, NORM_INF));
+}
+
+TEST_P(RMatViewTest, NonDefaultStepOutput) {
+    auto type = GetParam();
+    Mat mat(8,8,type);
+    cv::randn(mat, cv::Scalar::all(127), cv::Scalar::all(40));
+    Mat bigMat = Mat::zeros(16,16,type);
+    Mat out = bigMat(cv::Rect{4,4,8,8});
+    View view(cv::descr_of(out), out.ptr(), out.step);
+    auto viewMat = asMat(view);
+    Mat ref;
+    cv::Size ksize{1,1};
+    cv::blur(mat, viewMat, ksize);
+    cv::blur(mat, ref,     ksize);
+    EXPECT_EQ(0, cvtest::norm(ref, out, NORM_INF));
+}
+
+INSTANTIATE_TEST_CASE_P(Test, RMatViewTest,
+                        Values(CV_8UC1, CV_8UC3, CV_32FC1));
+
+struct RMatViewCallbackTest : public ::testing::Test {
+    RMatViewCallbackTest()
+        : mat(8,8,CV_8UC1), view(cv::descr_of(mat), mat.ptr(), mat.step1(), [this](){ callbackCalls++; }) {
+        cv::randn(mat, cv::Scalar::all(127), cv::Scalar::all(40));
+    }
+    int callbackCalls = 0;
+    Mat mat;
+    View view;
+};
+
+TEST_F(RMatViewCallbackTest, MoveCopy) {
+    {
+        View copy(std::move(view));
+        cv::util::suppress_unused_warning(copy);
+        EXPECT_EQ(callbackCalls, 0);
+    }
+    EXPECT_EQ(callbackCalls, 1);
+}
+
+static int firstElement(const View& view) { return *view.ptr(); }
+static void setFirstElement(View& view, uchar value) { *view.ptr() = value; }
+
+TEST_F(RMatViewCallbackTest, MagazineInteraction) {
+    cv::gimpl::magazine::Class<View> mag;
+    constexpr int rc = 1;
+    constexpr uchar value = 11;
+    mag.slot<View>()[rc] = std::move(view);
+    {
+        auto& mag_view = mag.slot<View>()[rc];
+        setFirstElement(mag_view, value);
+        auto mag_el = firstElement(mag_view);
+        EXPECT_EQ(mag_el, value);
+    }
+    {
+        const auto& mag_view = mag.slot<View>()[rc];
+        auto mag_el = firstElement(mag_view);
+        EXPECT_EQ(mag_el, value);
+    }
+    EXPECT_EQ(callbackCalls, 0);
+    mag.slot<View>().erase(rc);
+    EXPECT_EQ(callbackCalls, 1);
```

Comment:
All `_EQ` checks in GoogleTests have this parameter order: `(expected, actual)`.

Reply:
Will fix that in future updates, thanks!